src/postamble.js

```diff
@@ -175,8 +175,14 @@ Module['callMain'] = Module.callMain = function callMain(args) {
 #endif
       return;
     } else {
-      if (e && typeof e === 'object' && e.stack) Module.printErr('exception thrown: ' + [e, e.stack]);
-      throw e;
+      var toLog = e;
+      if (e && typeof e === 'object' && e.stack) {
+        toLog = [e, e.stack];
+      }
+      Module.printErr('exception thrown: ' + toLog);
+      // In a shell environment, callMain is ran from a promise context. Exceptions thrown from
```

Comment:
I don't think we need to mention the shell. Also callMain is not always run from a promise (depends on options). I'd just remove the comment - calling quit is self-explanatory, the application is now quitting?